gammapy/maps/wcsnd.py

```diff
@@ -589,6 +585,63 @@ def get_spectrum(self, region=None, func=np.nansum, weights=None):
 
         return self.to_region_nd_map(region=region, func=func, weights=weights)
 
+    def binary_erosion(self, margin, from_edges=False):
```

Comment:
My proposal would be to make this more like `WcsNDMap.smooth()`, so provide a `width` and `kernel` or `structure` argument and offer the options `box` and `disk`. Also allow for additional keyword forwarding to `binary_erosion()` for the boundary handling.

Reply:
- I renamed margin to width
- about the choice of option between `structure` or `kernel`, and if all `binary_erosion()` keywords are relevant for forwarding, I need to think about it, so maybe I could do that latter in a separate PR.